src/rust/src/lib.rs

```diff
@@ -221,6 +221,17 @@ mod _rust {
                     openssl_mod.add("_legacy_provider_loaded", false)?;
                 }
             }
+            cfg_if::cfg_if! {
+                if #[cfg(CRYPTOGRAPHY_OPENSSL_320_OR_GREATER)] {
+                    use std::ptr;
+                    if let Ok(available) = std::thread::available_parallelism() {
+                        // SAFETY: This sets a libctx provider limit, but we always use the same libctx by passing NULL.
+                        unsafe {
+                            openssl_sys::OSSL_set_max_threads(ptr::null_mut(), available.get() as u64);
+                        }
+                    }
```

Comment:
Sigh, this is the line that's uncovered, the else branch. But of course available_parallelism never fails in CI... not sure what the right thing to do is.

Rewrite the code somehow?

Also, what happens if someone else has called `OSSL_set_max_threads` (e.g., in theory, CPython?)